### PR TITLE
ETW Log Exporter

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

This PR adds ETW Log exporter using the current preview Logger API. 

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
Removing WIP, as it is ready for review.

File: exporters/etw/include/opentelemetry/exporters/etw/etw_config.h [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+ */
+typedef struct
+{
+  bool enableTraceId;            // Set `TraceId` on ETW events
+  bool enableSpanId;             // Set `SpanId` on ETW events
+  bool enableActivityId;         // Assign `SpanId` to `ActivityId`
+  bool enableActivityTracking;   // Emit TraceLogging events for Span/Start and Span/Stop Not used
+                                 // for Logs
+  bool enableRelatedActivityId;  // Assign parent `SpanId` to `RelatedActivityId`
+  bool enableAutoParent;  // Start new spans as children of current active span, Not used for Logs
```
Comment (ThomsonTan) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_config.h:38]: nit: keep alignment of comments?

File: exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h [hunk 2]
```diff
@@ -121,12 +121,22 @@
 #  define ETW_FIELD_STATUSCODE      "StatusCode"            /* Span status code        */
 #  define ETW_FIELD_STATUSMESSAGE   "StatusMessage"         /* Span status message     */
 #  define ETW_FIELD_SUCCESS         "Success"               /* Span success            */
+#  define ETW_FIELD_TIMESTAMP        "Timestamp"            /* Log timestamp   */
```
Comment (ThomsonTan) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h:124]:
```suggestion
#  define ETW_FIELD_TIMESTAMP       "Timestamp"            /* Log timestamp   */
```

File: exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h [hunk 3]
```diff
@@ -121,12 +121,22 @@
...
 
 /* Value constants */
 #  define ETW_VALUE_SPAN            "Span"                  /* ETW event name for Span */
+#  define ETW_VALUE_LOG             "Log"                  /* ETW event name for Log */
```
Comment (ThomsonTan) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h:128]:
```suggestion
#  define ETW_VALUE_LOG             "Log"                   /* ETW event name for Log */
```

File: exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h [hunk 4]
```diff
@@ -121,12 +121,22 @@
...
+
 #  define ETW_VALUE_SPAN_START      "SpanStart"             /* ETW for Span Start      */
 #  define ETW_VALUE_SPAN_END        "SpanEnd"               /* ETW for Span Start      */
 
+
+/* Log specific */
+# define  ETW_FIELD_LOG_BODY              "body"                 /* Log body   */
+# define ETW_FIELD_LOG_SEVERITY_TEXT      "severityText"         /* Sev text  */
+# define ETW_FIELD_LOG_SEVERITY_NUM       "severityNumber"          /* Sev num   */
```
Comment (ThomsonTan) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h:137]:
```suggestion
# define ETW_FIELD_LOG_SEVERITY_NUM       "severityNumber"      /* Sev num   */
```

File: exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h [hunk 5]
```diff
@@ -121,12 +121,22 @@
...
 /* Value constants */
 #  define ETW_VALUE_SPAN            "Span"                  /* ETW event name for Span */
+#  define ETW_VALUE_LOG             "Log"                  /* ETW event name for Log */
+
 #  define ETW_VALUE_SPAN_START      "SpanStart"             /* ETW for Span Start      */
 #  define ETW_VALUE_SPAN_END        "SpanEnd"               /* ETW for Span Start      */
 
+
+/* Log specific */
+# define  ETW_FIELD_LOG_BODY              "body"                 /* Log body   */
```
Comment (ThomsonTan) [inline exporters/etw/include/opentelemetry/exporters/etw/etw_fields.h:135]:
```suggestion
# define ETW_FIELD_LOG_BODY               "body"                 /* Log body   */
```

File: exporters/etw/include/opentelemetry/exporters/etw/etw_config.h (hunk 1 above)
Comment (lalitb) [reply exporters/etw/include/opentelemetry/exporters/etw/etw_config.h:38]: clang format tools format this way to keep under configured line limit :)

https://github.com/open-telemetry/opentelemetry-cpp/blob/2f462a5de1b85aaa42f53fa7fd7f69b4965c27b7/.clang-format#L11

**Pull request merged**